test/jdk/sun/net/www/protocol/https/ChunkedOutputStream.java

```diff
@@ -57,100 +79,99 @@
+        }  else if (path.equals("/d7")) {
+            count = 4;
+        }  else if (path.equals("/d8")) {
+            count = 5;
+        }
 
-            switch (count) {
+        switch (count) {
             case 0: /* test1 -- keeps conn alive */
             case 1: /* test2 -- closes conn */
-                String reqbody = req.getRequestEntityBody();
+
+                String reqbody = "";
+                try(InputStream inputStream = req.getRequestBody()) {
+                    if(inputStream != null) {
+                        reqbody = new String(inputStream.readAllBytes());
+                    }
+                }
                 if (!reqbody.equals(str1)) {
-                    req.sendResponse(500, "Internal server error");
-                    req.orderlyClose();
+                    req.sendResponseHeaders(500, -1);
+                    break;
                 }
-                String chunk = req.getRequestHeader("Transfer-encoding");
+                String chunk = req.getRequestHeaders().getFirst("Transfer-encoding");
                 if (!"chunked".equals(chunk)) {
-                    req.sendResponse(501, "Internal server error");
-                    req.orderlyClose();
+                    req.sendResponseHeaders(501, -1);
+                    break;
                 }
-                req.setResponseEntityBody(reqbody);
                 if (count == 1) {
-                    req.setResponseHeader("Connection", "close");
+                    req.getResponseHeaders().set("Connection", "close");
                 }
-                req.sendResponse(200, "OK");
-                if (count == 1) {
-                    req.orderlyClose();
+                req.sendResponseHeaders(200, 0);
+                try (PrintWriter pw = new PrintWriter(req.getResponseBody())) {
+                    pw.print(reqbody);
                 }
                 break;
             case 2: /* test 3 */
-                reqbody = req.getRequestEntityBody();
+                reqbody = new String(req.getRequestBody().readAllBytes());
                 if (!reqbody.equals(str2)) {
-                    req.sendResponse(500, "Internal server error");
-                    req.orderlyClose();
+                    req.sendResponseHeaders(500, -1);
+                    break;
                 }
-                int clen = Integer.parseInt (
-                        req.getRequestHeader("Content-length"));
+                int clen = Integer.parseInt (req.getRequestHeaders().getFirst("Content-length"));
                 if (clen != str2.length()) {
-                    req.sendResponse(501, "Internal server error");
-                    req.orderlyClose();
+                    req.sendResponseHeaders(501, -1);
+                    break;
+                }
+                req.getResponseHeaders().set("Connection", "close");
+                req.sendResponseHeaders(200, 0);
+                try (PrintWriter pw = new PrintWriter(req.getResponseBody())) {
```

Comment:
It's dangerous to rely on the default charset when using PrintWriter.
I'd suggest to change the client and server logic to use UTF-8 explicitly everywhere (when creating new strings, when obtaining string bytes, when using print writers or print streams, etc...)